src/main/java/com/itechart/javalab/library/dao/factory/impl/SqlDAOFactoryImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.itechart.javalab.library.dao.factory.impl;
+
+import com.itechart.javalab.library.dao.BookDao;
+import com.itechart.javalab.library.dao.conn.ConnectionPool;
+import com.itechart.javalab.library.dao.factory.DAOFactory;
+import com.itechart.javalab.library.dao.impl.SqlBookDaoImpl;
+
+public class SqlDAOFactoryImpl implements DAOFactory {
+
+    private static volatile SqlDAOFactoryImpl instance;
```

Comment:
why do you use `volatile` key word, if the singleton initialization is thread-safe?